autosklearn/automl.py

```diff
@@ -917,6 +944,144 @@ def refit(self, X, y):
         self._can_predict = True
         return self
 
+    def fit_pipeline(
+        self,
+        X: SUPPORTED_FEAT_TYPES,
+        y: SUPPORTED_TARGET_TYPES,
+        task: int,
+        is_classification: bool,
+        config: Configuration,
+        dataset_name: Optional[str] = None,
+        X_test: Optional[SUPPORTED_FEAT_TYPES] = None,
+        y_test: Optional[SUPPORTED_TARGET_TYPES] = None,
+        **kwargs: Dict,
+    ) -> Tuple[Optional[BasePipeline], RunInfo, RunValue]:
+        """ Fits and individual pipeline configuration and returns
+        the result to the user.
+
+        The Estimator constraints are honored, for example the resampling
+        strategy, or memory constraints, unless directly provided to the method.
+        By default, this method supports the same signature as fit(), and any extra
+        arguments are redirected to the TAE evaluation function, which allows for
+        further customization while building a pipeline.
+
+        Parameters
+        ----------
+        X: array-like, shape = (n_samples, n_features)
+            The features used for training
+        y: array-like
+            The labels used for training
+        X_test: Optionalarray-like, shape = (n_samples, n_features)
+            If provided, the testing performance will be tracked on this features.
+        y_test: array-like
+            If provided, the testing performance will be tracked on this labels
+        config: Configuration
+            A configuration object used to define a pipeline steps
+        dataset_name: Optional[str]
+            A string to tag and identify the Auto-Sklearn run
+        is_classification: bool
+            Whether the task is for classification or regression. This affects
+            how the targets are treated
+
+        Returns
+        -------
+        pipeline: Optional[BasePipeline]
+            The fitted pipeline. In case of failure while fitting the pipeline,
+            a None is returned.
+        run_info: RunInFo
+            A named tuple that contains the configuration launched
+        run_value: RunValue
+            A named tuple that contains the result of the run
+        """
+
+        # Get the configuration space
+        # This also ensures that the Backend has processed the
+        # dataset
+        if self.configuration_space is None:
+            self.configuration_space = self.fit(
+                X=X, y=y, task=task,
+                dataset_name=dataset_name if dataset_name is not None else self._dataset_name,
+                X_test=X_test,
+                y_test=y_test,
+                feat_type=kwargs.pop('feat_type', self._feat_type),
+                only_return_configuration_space=True)
+
+        # We do not want to overwrite existing runs
+        self.num_run += 1
+        config.config_id = self.num_run
```

Comment:
What do you think about allowing dictionaries, too? This would only require adding
```
        if isinstance(config, dict):
            config = Configuration(self.configuration_space, config)
```
and updating the signature above, but would make the whole thing a lot more versatile.

Reply:
tanks, I will add this